File: leetcode/problems/cpp/sort-array-by-increasing-frequency.cpp
// by @codeAbinash

bool comparePair(pair<int, int> &a, pair<int, int> &b) {
    if (a.second == b.second) return a.first > b.first;
    return a.second < b.second;
}

class Solution {
   public:
    vector<int> frequencySort(vector<int>& nums) {
        unordered_map<int, int> mp;
        int size = nums.size();
        for (int i = 0; i < size; i++) mp[nums[i]]++;

        vector<pair<int, int>> freq;
        for (auto& elem : mp) freq.push_back(elem);

        sort(freq.begin(), freq.end(), comparePair);

        vector<int> result;
        
        for (auto& elem : freq)
            for (int i = 0; i < elem.second; i++) result.push_back(elem.first);
        return result;
    }
};

File: leetcode/problems/java/sort-array-by-increasing-frequency.java
class Solution {
  public int[] frequencySort(int[] nums) {
    int count[] = new int[201];
    for (int n : nums)
      count[n + 100]++;
    for (int i = nums.length - 1; i >= 0;) {
      int m = 0, ind = -1;
      for (int j = 0; j < 201; j++) {
        if (count[j] > m) {
          m = count[j];
          ind = j;
        }
      }
      for (int j = 0; j < m; j++)
        nums[i--] = ind - 100;
      count[ind] = 0;
    }
    return nums;
  }
}